src/coreclr/src/pal/src/misc/sysinfo.cpp

```diff
@@ -221,6 +225,8 @@ GetSystemInfo(
     lpSystemInfo->lpMaximumApplicationAddress = (PVOID) VM_MAXUSER_ADDRESS;
 #elif defined(__linux__)
     lpSystemInfo->lpMaximumApplicationAddress = (PVOID) (1ull << 47);
+#elif defined(__sun)
+    lpSystemInfo->lpMaximumApplicationAddress = (PVOID) 0xfffffd7fffe00000ul;
```

Comment:
This value was copied from https://github.com/illumos/illumos-gate/blob/4e0c5eff9af325c80994e9527b7cb8b3a1ffd1d4/usr/src/uts/i86pc/sys/machparam.h#L235, unfortunately that header is not included in the base box. After some discussion in #illumos channel over at freenode IRC, we may get a helper function like `getusrstack()` (similar to `getpagesize()`) in the future, rather than a hardcoded value macro.

Reply:
So on SunOS, the user code can live even in such a high address range?

Reply:
Hi @janvorli, that's correct (at least as far as illumos is concerned, I work on that but cannot speak for Oracle Solaris). Though this value can shift a bit. For example, on my system it's currently `0xfffffc7fffe00000` and that's because @am11 used a slightly older base of illumos where we had to shift that due to part of our KPTI mitigations.  However, it is different in different situations. For example if running inside of Xen pv (though that's rather uncommon) that today is `0x00007fffffe00000`, e.g. below the VA hole. Those are values on x86. On SPARC right now the user limit is `0xFFFFFFFF80000000`.

I'm looking at adding a sysconf style interface to get the maximum mapable address. I suspect on other systems with the advent of 5-level paging the maximum address may shift, though that may not matter for most consumers of the CLR. I would suggest limiting it at the VA hole on amd64 until such time as we have that available and falling back to the VA hole when not available.